### PR TITLE
rox 13015 collector integration tests should check for process info in endpoints and find it

### PR DESCRIPTION
**Pull request by JoukoVirtanen**
## Description

The ConnScraper integration test which checks endpoint information sent by collector will now check that the endpoint information includes the process originator. This PR is related to https://github.com/stackrox/collector/pull/868

The difference between the two PRs is that this one will pass once Collector reports process data in the endpoints and the one linked to will pass if Collector doesn't report process data. This means that integration tests here will fail. See "Testing Performed" to see how to test this locally.

## Checklist
- [ ] Investigated and inspected CI test results

**Automated testing**
  - [x] Added integration tests

## Testing Performed

git checkout ovalenti/ROX-12803-add-process-info-to-network-endpoints
make clean
make
git checkout jv-rox-13015-collector-integration-tests-should-check-for-process-info-in-endpoints-and-find-it
make
make integration-tests-connscraper

Checking out the first branch and building the collector there make it so that the tests are run using the branch that adds process information to endpoints.

### CODE REVIEW

File: integration-tests/endpoint.go [hunk 1]
```diff
@@ -9,18 +9,20 @@ type EndpointInfo struct {
...
-		return nil, fmt.Errorf("invalid gRPC string for endpoint info: %s", line)
+	if len(parts) != 5 {
+		return nil, fmt.Errorf("Invalid gRPC string for endpoint info: %s", line)
 	}
 
 	return &EndpointInfo {
 		Protocol: parts[1],
 		ListenAddress: parts[2],
 		CloseTimestamp: parts[3],
+		Originator: parts[4],
```
Comment (Stringy) [inline integration-tests/endpoint.go:26]: Can we turn Originator into a `ProcessInfo`? I appreciate the string format is different but it contains the same data and would make the assertion in the test a little clearer 

I guess the ideal would be to store the information in the same format in the Process bucket as well as in the Endpoints bucket so they can be parsed in the same way

Comment (JoukoVirtanen) [reply integration-tests/endpoint.go]: Done

File: integration-tests/process_originator.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+func removeQuotesAndWhiteSpace(s string) string {
+	s = strings.TrimSpace(s)
+	if len(s) > 0 && s[0] == '"' {
+		s = s[1:]
+	}
+	if len(s) > 0 && s[len(s)-1] == '"' {
+		s = s[:len(s)-1]
+	}
```
[the remark this answers is not in the document]
Comment (JoukoVirtanen) [reply integration-tests/process_originator.go:22]: I am not sure that we can use Trim here. I think Trim can remove multiple quotation marks and we just want to remove the first and the last.

File: integration-tests/process_originator.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	return s
+}
+
+func NewProcessOriginator(line string) (*ProcessOriginator, error) {
+	if line == "<nil>\n" {
+		return nil, fmt.Errorf("ProcessOriginator is nil")
+	}
+
+	r := regexp.MustCompile("process_name:(.*)process_exec_file_path:")
```
[the remark this answers is not in the document]
Comment (JoukoVirtanen) [reply integration-tests/process_originator.go]: Done

File: integration-tests/process_originator.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+	processExecFilePath := removeQuotesAndWhiteSpace(processExecFilePathArr[1])
+
+	r = regexp.MustCompile("process_args:(.*)\n$")
+	processArgsArr := r.FindStringSubmatch(line)
+	var processArgs string
+	if len(processArgsArr) !=2 {
+		processArgs = ""
+	} else {
+		processArgs = removeQuotesAndWhiteSpace(processArgsArr[1])
+	}
```
[the remark this answers is not in the document]
Comment (JoukoVirtanen) [reply integration-tests/process_originator.go]: Done

File: integration-tests/endpoint.go [hunk 5]
```diff
@@ -9,18 +9,22 @@ type EndpointInfo struct {
...
 func NewEndpointInfo(line string) (*EndpointInfo, error) {
 	parts := strings.Split(line, "|")
 
-	if len(parts) != 4 {
-		return nil, fmt.Errorf("invalid gRPC string for endpoint info: %s", line)
+	if len(parts) != 5 {
+		return nil, fmt.Errorf("Invalid gRPC string for endpoint info: %s", line)
 	}
 
+	originator, _ := NewProcessOriginator(parts[4])
```
[the remark this answers is not in the document]
Comment (JoukoVirtanen) [reply integration-tests/endpoint.go]: Done

**Review by Stringy — APPROVED** [review]

**Pull request closed without merging**